### ISSUE TITLE
ScrollBox: Upper shadow doesn't always hide at top of extent

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#159]
The upper shadow configurable for scrollbox under some circumstances won't always be hidden once the user scrolls back to the top of the frame.

This can be seen with the demo code and video from #158, copied below:

```lua
local DataProvider = CreateDataProviderIndexRange(20);

local ScrollView = CreateScrollBoxListLinearView();
ScrollView:SetDataProvider(DataProvider);
ScrollView:SetElementExtent(28);
ScrollView:SetElementInitializer("Button", "", nop);

local ScrollBox = CreateFrame("Frame", nil, UIParent, "WowScrollBoxList");
ScrollBox:SetPoint("CENTER");
ScrollBox:SetSize(400, 400);
ScrollBox:SetLowerShadowAtlas("colorblind-bar-red", true);
ScrollBox:SetUpperShadowAtlas("colorblind-bar-red", true);

local ScrollBar = CreateFrame("EventFrame", nil, UIParent, "WowTrimScrollBar");
ScrollBar:SetPoint("TOPLEFT", ScrollBox, "TOPRIGHT");
ScrollBar:SetPoint("BOTTOMLEFT", ScrollBox, "BOTTOMRIGHT");

ScrollBox.Background = ScrollBox:CreateTexture(nil, "BACKGROUND");
ScrollBox.Background:SetAllPoints(ScrollBox);
ScrollBox.Background:SetColorTexture(0, 0, 0);

ScrollUtil.InitScrollBoxWithScrollBar(ScrollBox, ScrollBar, ScrollView);
```

In the original video below we can see that, from 5 seconds in, when dragging the scrollbar thumb from the bottom to the top of the frame the top shadow atlas doesn't hide. This later corrects itself when we scroll down and instead scroll up via the mousewheel.

https://user-images.githubusercontent.com/287102/128736432-be9bc40e-039a-4fe1-94ac-a278d9f5ac55.mp4

When the issue occurs, the scroll percentage appears to be a very small value as shown below - so it's likely that [this condition](https://github.com/Gethe/wow-ui-source/blob/fb934dc6558497b2a494898a33e0587a0e6bfe3d/SharedXML/Scroll/ScrollBox.lua#L158) for the upper shadow visibility needs to instead check if the scroll percentage is beyond MathUtil.Epsilon instead of flat zero.

![image](https://user-images.githubusercontent.com/287102/128739281-3ca7d095-1314-43ff-8ee9-f7744f9e8325.png)

**Comment by Meorawr** [comment]
Fixed in 2.5.2.39801 for BCC; will wait until fixed on retail before closing.

**Comment by Meorawr** [comment]
Fixed in 1.14.0.39958 and 9.1.5.39977.
